Contributing.md

```diff
@@ -1,30 +1,472 @@
+3. To keep the code **open** (as in free).
+4. To keep the code open (as in **encouraging**). If we achieve #2, this comes for free. But have fun doing it. 
+
+These guidelines are split up into a few main topics:
+
+* [The Release Process](#the-tari-release-process). This explains how the release schedule for Tari works post-mainnet 
+  launch. 
+* The [Feature RFC process](#feature-rfc-process) describes how new features get into Tari, from conception 
+  to implementation.
+* [Pull requests](#pull-requests). This section offers guidelines on how to get your code merged into the code base 
+  with the minimum of fuss. 
+* [Code reviews](#code-reviews) are integral to keeping the code secure and performant. This section 
+  offers tips for effective preventing bugs from scuttling past your gimlet gaze. 
+* Bug bounty and disclosure policy.
+
+# The Tari release process
+
+The Tari release process draws inspiration from the Rust [compiler release process] and borrows some of its key 
+ideas.
+
+The process is well-suited for a large open-source project that desires a regular release schedule but also needs 
+the flexibility of having features being "in development" over several cycles.
+
+## Release branches
+
+Post mainnet, there are 4 release branches in the Tari repo:
+
+* `mainnet` - the code on this branch represents the latest "official release". Almost every commit on this branch 
+  will be tagged with a version number.
+* `development` - this is the cuttingest-of-edge branch. Almost all new code comes in via the `development` branch.
+* `nextnet` - the code on this branch will become the `mainnet` branch on the next release. Feature code from 
+  `dibbler` will be present here, but if they are not yet ready for release, they will be disabled behind a feature 
+  gate.
+* `dibbler` - this branch contains experimental features that might not work, be rigorously tested, or even break 
+  consensus if the code were to run on mainnet.
+
+Almost all new code enters the codebase via the `development` branch. Hotfixes and urgent updates can be made on 
+other branches from time to time. These changes will then be cherry-picked into the other branches to keep 
+the code base consistent.
+
+## Testnets
+
+There will also be three testnets in addition to mainnet:
+
+* `mainnet` - the "real-money" network.
+* `stagenet` - a testnet version of mainnet. Nodes on stagenet should all be running code built from the `mainnet`
+  branch to replicate mainnet as close as possible. The base node version on `stagenet` will always match
+  the latest `mainnet` release. Network resets (i.e. a new genesis block) should be very rare.
+* `nextnet` - nodes on nextnet should be running code compiled from the `nextnet` branch. This is the last chance to
+  battle-test new features before they hit mainnet. Network resets should be somewhat rare.
+* `dibbler` - The dibbler network is the most experimental of the testnets. Things can break here. We hope they 
```

Comment:
Thinking about this more. If the answer is "yes" then we probably need extra flags for where certain PR's need to get merged. Or- potentially a single fix requires two PR's. One into development, one into dibbler/nextnet etc.

If we only merge into development someone has to know they now need to cherry-pick somewhere else. It creates work, and the possibility that work never happens. Once identified as a fix for a specific net, the requirement to open two PR's moves the actions back to the developer making the fix, and CI/Bots to take care of the merging etc.

Reply:
The code will migrate to nextnet. But it won't be active until the feature it is gated behind is stabilised.

Reply:
Some PRs are bug fixes, or don't break consensus, or are otherwise too small to necessitate a feature gate. These PRs will get merged into development, and migrate to mainnet over the next 3 releases.

Other PRs that will need to be behind a gate from the point it's merged development.